lib/modules.nix

```diff
@@ -474,26 +488,58 @@ rec {
           [{ inherit (module) file; inherit value; }]
         ) configs;
 
+      # Convert an option tree decl to a submodule option decl
+      optionTreeToOption = decl:
+        if isOption decl.options
+        then decl
+        else decl // {
+            options = mkOption {
+              type = types.submoduleWith {
+                modules = [ { options = decl.options; } ];
+                shorthandOnlyDefinesConfig = null;
```

Comment:
This should have a comment explaining the reasoning behind this: How it's a bit hacky but needed as a default, that `null` is only an internally valid value, that it's only safe here because of the merging function patch and because it's always going to be merged with at least one other module.

In the future this might become obsolete with https://github.com/NixOS/nixpkgs/issues/162398

Reply:
Added

```
                # `null` is not intended for use by modules. It is an internal
                # value that means "whatever the user has declared elsewhere".
                # This might become obsolete with https://github.com/NixOS/nixpkgs/issues/162398
```